### PR TITLE
Rewrite visualization for age_gender and emotion recognition & fixed bug with pose_estimation recognition

### PR DESCRIPTION
**Pull request by AlexanderSlav**

### CODE REVIEW

File: modelplace_api/visualization.py [hunk 1]
```diff
@@ -67,6 +67,37 @@ def draw_line(
...
 
+def draw_text_label(
+    image: np.ndarray,
+    detection: AgeGenderLabel,
+    text: str,
+    font_scale: float = 0.6,
+    thickness: int = 1,
+    bg_color: tuple = (128, 128, 128),
+) -> np.ndarray:
+    font = cv2.FONT_HERSHEY_TRIPLEX
```
Comment (lxdv) [inline modelplace_api/visualization.py:78]: Let's use it as a default argument

File: modelplace_api/visualization.py [hunk 2]
```diff
@@ -67,6 +67,37 @@ def draw_line(
...
+        (int(text_offset_x + text_width + 2), int(text_offset_y - text_height - 2)),
+    )
+    cv2.rectangle(image, box_coords[0], box_coords[1], bg_color, cv2.FILLED)
+    cv2.putText(
+        image,
+        text,
+        (int(text_offset_x), int(text_offset_y)),
+        font,
+        fontScale=font_scale,
+        color=(255, 255, 255),
```
Comment (lxdv) [inline modelplace_api/visualization.py:95]: Can we use it as a `text_color` default variable as we do with `bg_color`?

File: modelplace_api/visualization.py [hunk 3]
```diff
@@ -67,6 +67,37 @@ def draw_line(
...
+    font_scale: float = 0.6,
+    thickness: int = 1,
+    bg_color: tuple = (128, 128, 128),
+) -> np.ndarray:
+    font = cv2.FONT_HERSHEY_TRIPLEX
+    (text_width, text_height) = cv2.getTextSize(
+        text, font, fontScale=font_scale, thickness=thickness,
+    )[0]
+    text_offset_x = detection.bbox.x1
+    text_offset_y = detection.bbox.y1 - 10
```
Comment (lxdv) [inline modelplace_api/visualization.py:83]: Looks like an offset variable `10` also can be specified as a default argument

File: modelplace_api/visualization.py [hunk 4]
```diff
@@ -286,14 +317,14 @@ def draw_pose_estimation_result(
                     image_with_skeletons,
                     (int(link.joint_a.x), int(link.joint_a.y)),
                     (int(link.joint_b.x), int(link.joint_b.y)),
-                    RGB_COLORS[-1][::-1],
+                    RGB_COLORS[-1][2::-1],
```
Comment (lxdv) [inline modelplace_api/visualization.py:320]: Let's rename `RGB_COLORS` to `RGBA_COLORS` for a better understanding of why we do this way

File: modelplace_api/visualization.py [hunk 5]
```diff
@@ -490,80 +521,50 @@ def draw_classification_result(
...
+    images.append(source_image)
     for detection in detections:
-        image = image_with_bbox.copy()
-        image = cv2.rectangle(
-            image,
+        image_with_boxes = cv2.rectangle(
+            image_with_boxes,
             (int(detection.bbox.x1), int(detection.bbox.y1)),
             (int(detection.bbox.x2), int(detection.bbox.y2)),
             RGB_COLORS[196],
```
Comment (lxdv) [inline modelplace_api/visualization.py:533]: `bg_color`?

File: modelplace_api/visualization.py [hunk 6]
```diff
@@ -490,80 +521,50 @@ def draw_classification_result(
...
-        }
-        classification_labels["Age"] = detection.age
-        image = draw_classification_legend(
-            image,
-            classification_labels,
-            font_scale=0.75,
-            font_thickness=2,
-            height_offset=20,
-            cell_height=50,
+        bg_color = RGB_COLORS[196]
```
Comment (lxdv) [inline modelplace_api/visualization.py:536]: Maybe better to specify as a default argument? 

@PVSemk what do you think?

File: modelplace_api/visualization.py [hunk 7]
```diff
@@ -490,80 +521,50 @@ def draw_classification_result(
...
+    images.append(source_image)
     for detection in detections:
-        image = image_with_bbox.copy()
-        image = cv2.rectangle(
-            image,
+        image_with_boxes = cv2.rectangle(
+            image_with_boxes,
             (int(detection.bbox.x1), int(detection.bbox.y1)),
             (int(detection.bbox.x2), int(detection.bbox.y2)),
             RGB_COLORS[196],
```
Comment (lxdv) [inline modelplace_api/visualization.py:559]: bg_color?

File: modelplace_api/visualization.py [hunk 8]
```diff
@@ -490,80 +521,50 @@ def draw_classification_result(
...
             thickness=8,
```
Comment (lxdv) [inline modelplace_api/visualization.py:560]: Maybe better to specify as a default argument?

@PVSemk what do you think?

File: modelplace_api/visualization.py [hunk 9]
```diff
@@ -490,80 +521,50 @@ def draw_classification_result(
...
```
Comment (lxdv) [inline modelplace_api/visualization.py:534]: Maybe better to specify as a default argument?

@PVSemk what do you think?

Comment (PVSemk) [reply modelplace_api/visualization.py:534]: Resonable, as we did it earlier: [link](https://github.com/opencv-ai/modelplace-api/pull/9/files/fd65183bc05a04afed2764aaeb8fdbe804490ddd#diff-63fde0013d268f8f6319418745f702a12000cff7c559b8c3347b6eed32128077R75)

File: modelplace_api/visualization.py (hunk 6 above)
Comment (PVSemk) [reply modelplace_api/visualization.py:536]: Reasonable, we did it earlier: [link](https://github.com/opencv-ai/modelplace-api/pull/9/files/fd65183bc05a04afed2764aaeb8fdbe804490ddd#diff-63fde0013d268f8f6319418745f702a12000cff7c559b8c3347b6eed32128077R76)

File: modelplace_api/visualization.py (hunk 8 above)
Comment (PVSemk) [reply modelplace_api/visualization.py:560]: Resonable, as we did it earlier: [link](https://github.com/opencv-ai/modelplace-api/pull/9/files/fd65183bc05a04afed2764aaeb8fdbe804490ddd#diff-63fde0013d268f8f6319418745f702a12000cff7c559b8c3347b6eed32128077R75)

File: modelplace_api/visualization.py [hunk 10]
```diff
@@ -67,6 +67,37 @@ def draw_line(
             i += 1
 
 
+def draw_text_label(
+    image: np.ndarray,
+    detection: AgeGenderLabel,
+    text: str,
+    font_scale: float = 0.6,
+    thickness: int = 1,
+    bg_color: tuple = (128, 128, 128),
```
Comment (PVSemk) [inline modelplace_api/visualization.py:76]: Probably, usage of such parameters should be consistent: either use color from RGB_COLORS or manually coded RGB color in all appearances. However, I believe we won't find the exact (128, 128, 128) color in our array.

@lxdv, what do you think?

File: modelplace_api/visualization.py [hunk 11]
```diff
@@ -67,6 +67,37 @@ def draw_line(
             i += 1
 
 
+def draw_text_label(
+    image: np.ndarray,
+    detection: AgeGenderLabel,
```
Comment (PVSemk) [inline modelplace_api/visualization.py:72]: Shouldn't it be `Union[AgeGenderLabel, EmotionLabel]` there?
Is this function for these two models only?

File: modelplace_api/visualization.py (hunk 10 above)
Comment (lxdv) [reply modelplace_api/visualization.py]: Agree. Looks like we could also create one more array with not just palette colors but rather with specific design colors.

At the moment, @AlexanderSlav decided to use `bg_color: tuple = RGBA_COLORS[196]` so let's use it for now

File: modelplace_api/visualization.py [hunk 12]
```diff
@@ -67,6 +67,39 @@ def draw_line(
...
 
 
+def draw_text_label(
+    image: np.ndarray,
+    detection: Union[AgeGenderLabel, EmotionLabel],
+    text: str,
+    font_scale: float = 0.6,
+    thickness: int = 1,
+    bg_color: tuple = RGBA_COLORS[196][2::-1],
+    text_color: tuple = (255, 255, 255),
```
Comment (lxdv) [inline modelplace_api/visualization.py:77]: minor: @PVSemk probably we should use color from RGBA_COLORS as you said earlier

**Comment by lxdv** [comment]
All code refactoring will be done in a new PR

**Pull request merged**